dask_kubernetes/core.py

```diff
@@ -464,28 +509,14 @@ def __init__(
         )
         self._generate_name = escape(self._generate_name)
 
-        # Default labels that can't be overwritten
-        self.pod_template.metadata.labels["dask.org/cluster-name"] = self._generate_name
-        self.pod_template.metadata.labels["user"] = escape(getpass.getuser())
-        self.pod_template.metadata.labels["app"] = "dask"
-        self.pod_template.metadata.namespace = self._namespace
-
-        if self.env:
-            self.pod_template.spec.containers[0].env.extend(
-                [
-                    kubernetes.client.V1EnvVar(name=k, value=str(v))
-                    for k, v in self.env.items()
-                ]
-            )
-        self.pod_template.metadata.generate_name = self._generate_name
+        self._fill_pod_templates()
```

Comment:
Again I see the need for breaking this out into a function but it's a bit magic.